CLI/turBOT.py

```diff
@@ -26,81 +37,96 @@ def progress(message):
-        t.start()
-        # p = subprocess.Popen(["<Path to Decompiler>", "<Disassemble Command>", source_dir, output_dir]) TODO: HOW DO WE ACTUALLY DISASSEMBLE STUFF??
-        time.sleep(5)
-        # p.wait() # Wait for subprocess to finish executin
-        # Condition Variable for "Loading" process
-        global done
-        done = True
-
-        t.join()
-
-        # TODO: GET RID OF THIS SECTION ONCE WE HAVE DISASSEMBLY ACTUALLY WORKING
-        with open(output_dir, 'w') as fp:
-            fp.write("This is a sample File")
-        
-        print(source_dir + " has been disassembled to " + output_dir)
+            print("Program is currently disassembling")
+        else:
+            print(f"{ERROR}ERROR: Disassembly target file not found{NO_COLOUR}")
+            return 2
+
+        args = ["ddisasm", source_dir, "--asm", output_dir] # Console Commands Go Here, split strings
+        p = subprocess.Popen(args, stdout=subprocess.PIPE)
+        p.wait() # Wait for subprocess to finish executing
+        exit_code = p.poll()
+        # Return exit code to know if to proceed
+        return exit_code
+
     except Exception as e:
         print(e)
-        print("An error occured during the decompilation proccess")
+        print(f"{ERROR}An error occured during the decompilation proccess{NO_COLOUR}")
+        # Return exception error code
+        return EXCEPTION_ERROR_CODE
 
 # Disassemble, Assmeble and Obfuscate are similar atm for demo purposes.
 # They will be moved to modules later on as they will get complex once integrated with their corresponding tools
 def reassemble(source_dir, output_dir): 
     try:
         if os.path.isfile(source_dir):
-            print("File found")
-        message = "Program is currently compiling"
-        t = threading.Thread(target=progress, args=(message,))
-        t.start()
-        # p = subprocess.Popen(["<Path to Compiler>", "<Assmeble Command>", source_dir, output_dir]) TODO: HOW DO WE ACTUALLY COMPILE STUFF??
-        time.sleep(5)
-        # p.wait() # Wait for subprocess to finish executin
-        # Condition Variable for "Loading" process
-        global done
-        done = True
-
-        t.join()
-
-        # TODO: GET RID OF THIS SECTION ONCE WE HAVE DECOMPILATION ACTUALLY WORKING
-        with open(output_dir, 'w') as fp:
-            fp.write("This is a sample File")
+            print("Program is currently reassembling")
+        else:
+            print(f"{ERROR}ERROR: Reassembly target file not found{NO_COLOUR}")
+            return 2
+        
+        args = ["g++", "-pthread", "-no-pie", "-o", output_dir, source_dir] # Console Commands Go Here, split strings
+        p = subprocess.Popen(args, stdout=subprocess.PIPE)
+        p.wait() # Wait for subprocess to finish executing
+        exit_code = p.poll()
+        # Return exit code to know if to proceed
+        return exit_code
         
-        print(source_dir + " has been compiled to " + output_dir)
```

Comment:
Do we want to remove all messages relating to printing if the function has successfully completed

Reply:
Do you mean the "program is currently reassembling" line, because I think it's important to have at least that much. Otherwise I think you're commenting on removed code

Reply:
Yeah I meant like when the operation has completed do we have a message